### DISCUSSION TITLE
推荐使用的正则表达式

### DISCUSSION THREAD

**Discussion by eric2788** [eric2788/bilibili-jimaku-filter#21] [想法]
这是我目前使用的

```regex
^(?<n>[^【】]+?)?\:?\s*【(?<cc>[^【】]+?)】?$
```

有其他更好的可以在下方提出。

**Comment by eric2788** [comment]
可捕捉括號

```regex
^(?<n>[^【】()]+?)?\:*\s*[【(](?<cc>[^【】()]+?)[】)]*$
```

**Comment by eric2788** [comment]
可捕捉歌詞彈幕發送機的彈幕，可捕捉格式：

- ❄【彈幕歌詞】❄
- ♪【音樂歌詞】~


```regex
^(?<n>[^【】]+?)?\:?\s*【(?<cc>[^【】]+?)(】.?)?$
```

更新 22:15 - 1/19-2021: 修復會吞掉最後一個字的問題

**Comment by KyriosChenn** [comment]
根据我的DD经验，直播间的文本大致可能出现：
1. 太anti了
2. 30【有了有了
3. 【这个这个】
4. 【原来如此
5. 】真的有这种人吗
6. 有在自己的满月直播里】
7. ♪【歌词
8. ♪【歌词】
9. ♪【歌词】♪

除去第一个是普通弹幕之外，其他的都是可能的同传文本。
所以我的正则是:
`^(?<n>[^【】]+?)?\:*\s*([【】]+?)(?<cc>[^【】]*)(】*)(.*)$`

应该能把这些情况覆盖到。但是由于宽容度增加了，有可能出现意外的情况（虽然现在还没碰到）。
